internal/services/iotcentral/parse/nested_item.go

```diff
@@ -0,0 +1,145 @@
+package parse
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/resourceids"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type NestedItemObjectType string
+
+const (
+	baseUrlTemplate                                 = "https://%s.azureiotcentral.com"
+	NestedItemTypeOrganization NestedItemObjectType = "organizations"
+)
+
+// PossibleNestedItemObjectTypeValues returns a string slice of possible "NestedItemObjectType" values.
+func PossibleNestedItemObjectTypeValues() []string {
+	return []string{string(NestedItemTypeOrganization)}
+}
+
+var _ resourceids.Id = NestedItemId{}
+
+type NestedItemId struct {
+	IotcentralBaseUrl string
+	SubDomain         string
+	NestedItemType    NestedItemObjectType
+	Id                string
+}
+
+func NewNestedItemID(iotcentralBaseUrl string, nestedItemType NestedItemObjectType, id string) (*NestedItemId, error) {
```

Comment:
so we can simplify this somewhat:

```suggestion
func NewNestedItemID(name string, domainSuffix string, nestedItemId string) (*NestedItemId, error) {
```

since we have all of this information when creating the object - meaning that we only need to be concerned with parsing the domain name into `name` and `domainSuffix` when the ID is being parsed